src/common/no-objects.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { isEmpty } from 'lodash'
+
+import propTypes from './prop-types-decorator'
+
+const NoObjects = ({ children, className, collection, message, predicate }) => !predicate(collection)
+  ? <h2 className={className}><img src='assets/loading.svg' /></h2>
+  : isEmpty(collection)
+    ? <p className={className}>{message}</p>
+    : <div>{children}</div>
+
+propTypes(NoObjects)({
+  children: propTypes.node.isRequired,
+  collection: propTypes.oneOfType([
+    propTypes.array,
+    propTypes.object
+  ]).isRequired,
+  message: propTypes.node.isRequired,
+  className: propTypes.string,
+  predicate: propTypes.func.isRequired
```

Comment:
I even think the `boolean` should not even be passed as a prop and the component should test `collection` directly.